textacy/texts.py

```diff
@@ -164,29 +185,132 @@ def as_bag_of_terms(self, weighting='tf', normalized=True, binary=False,
-        raise NotImplementedError()
+    def as_semantic_network(self, nodes='terms',
+                            edge_weighting='default', window_width=10):
+        """
+        Represent doc as a semantic network, where nodes are either 'terms' or
+        'sents', and edges between nodes may be weighted in different ways.
 
-    def as_semantic_network(self):
-        raise NotImplementedError()
+        Args:
+            nodes (str, {'terms', 'sents'}): type of doc component to use as nodes
+                in the semantic network
+            edge_weighting (str, optional): type of weighting to apply to edges
+                between nodes; if ``nodes == 'terms'``, options are {'cooc_freq', 'binary'},
+                if ``nodes == 'sents'``, options are {'cosine', 'jaccard'}; if
+                'default', 'cooc_freq' or 'cosine' will be automatically used
+            window_width (int, optional): size of sliding window over terms that
+                determines which are said to co-occur; only applicable if 'terms'
+
+        Returns:
+            :class:`networkx.Graph <networkx.Graph>`: where nodes represent either
+                terms or sentences in doc; edges, the relationships between them
+
+        .. seealso:: :func:`network.terms_to_semantic_network() <textacy.representations.network.terms_to_semantic_network>`
+        .. seealso:: :func:`network.sents_to_semantic_network() <textacy.representations.network.sents_to_semantic_network>`
+        """
+        if nodes == 'terms':
+            if edge_weighting == 'default':
+                edge_weighting = 'cooc_freq'
+            return network.terms_to_semantic_network(
+                list(self.words()), window_width=window_width,
+                edge_weighting=edge_weighting)
+        elif nodes == 'sents':
+            if edge_weighting == 'default':
+                edge_weighting = 'cosine'
+            return network.sents_to_semantic_network(
+                list(self.sents), edge_weighting=edge_weighting)
+        else:
+            msg = 'nodes "{}" not valid; must be in {}'.format(nodes, {'terms', 'sents'})
+            raise ValueError(msg)
+
+    def as_terms_list(self, words=True, ngrams=(2, 3), named_entities=True,
+                      dedupe=True, lemmatize=True, **kwargs):
+        """
+        Represent doc as a sequence of terms -- which aren't necessarily in order --
+        including words (unigrams), ngrams (for a range of n), and named entities.
+        NOTE: Despite the name, this is a generator function; to get a *list* of terms,
+        just wrap the call like ``list(doc.as_terms_list())``.
+
+        Args:
+            words (bool, optional): if True (default), include words in the terms list
+            ngrams (tuple(int), optional): include a range of ngrams in the terms list;
+                default is ``(2, 3)``, i.e. bigrams and trigrams are included; if
+                ngrams aren't wanted, set to False-y
```

Comment:
I know, this is slightly awkward naming, since words are in fact unigrams. However, ngrams usually refer to N >= 2, i.e. bigrams, trigrams, ... Since this distinction is used elsewhere (e.g. extract.words, extract.ngrams), I'm inclined to leave it in, unless you have strong objections...